detectors/aws/ecs/ecs.go

```diff
@@ -125,6 +129,28 @@ func (detector *resourceDetector) Detect(ctx context.Context) (*resource.Resourc
 			}
 		}
 
+		if accountId, err := detector.getAccountID(taskMetadata.TaskARN); err == nil {
```

Comment:
If those errors aren't nil, we should report them. Either with `otel.Handle()`, or with `span.RecordError()` (I'm thinking the latter is best).

Reply:
These error take place in the detector, which is run once on startup. Usually there won't be an active span there. In other SDKs iirc such errors are logged out, but I did not see that done in other detectors.

Reply:
I am not familiar with `otel.Handle()`, I'll look it up :-)

Reply:
@dmathieu I think the solution is much simpler, see c6f59b86f80c95ed926295eb419830763a3cbfc5 :-)